include/boost/next_prior.hpp

```diff
@@ -140,6 +138,24 @@ struct prior_impl1< T, Distance, true >
     }
 };
 
+template< typename T, typename Distance, bool IsIterator = is_iterator< T >::value >
+struct prior_advance_impl :
+    public prior_minus_assign_impl< T, Distance >
+{
+};
+
+template< typename T, typename Distance >
+struct prior_advance_impl< T, Distance, true >
+{
+    static T call(T x, Distance n)
+    {
+        // Avoid negating n to sidestep possible integer overflow
+        std::reverse_iterator< T > rx(x);
```

Comment:
The missing feature of `boost::next/prior` is tag dispatch based on Boost's IteratorTraversalCategory rather than Standard's IteratorCategory. The use of `std::reverse_iterator` forbids calling `boost::prior` with iterators like
```
auto twice = [](int x) { return x + x; };

std::vector<int> ints1 = ...;
auto it1 = boost::make_transform_iterator(ints1.begin(), twice);
// it1 is RandomAccessTraversalIterator in Boost's IteratorTraversalCategory.
// However, it is just InputIterator in Standard's IteratorCategory, because
// its reference type is not a real reference.

std::list<int> ints2 = ...;
auto it2 = boost::make_transform_iterator(ints2.begin(), twice);
// it2 is BidirectionalTraversalIterator in Boost's IteratorTraversalCategory.
// However, it is just InputIterator in Standard's IteratorCategory, because
// its reference type is not a real reference.
```
since they are just InputIterator. IMHO, supporting Boost's IteratorTraversalCategory is more important than preventing overflow.